### DISCUSSION TITLE
kitten launch nvim causes lsp issues

### DISCUSSION THREAD

**Discussion by dudicoco** [kovidgoyal/kitty#6908] [Q&A]
Hi,

I'm not sure if this is a bug within kitty or something related to my config so I opened it within the discussions rather than within the issues for now.

I'm launching neovim via the kitten launch command:
```
kitten @ launch --cwd ~/repositories/my-repo nvim
```

When opening any yaml file within neovim I get an error at the bottom:
```
Client 1 quit with exit code 127 and signal 0                                                                                                         
```

This error seems to be lsp related.
I'm not getting the same error when starting nvim normally without the kitten launch command.

Any idea what could be causing this?

**Comment by kovidgoyal** [comment]
It will be related to something in your shell rc files, probably PATH. 
When you run nvim directly your shell rc files are not sourced and any
changes they make to the environmnet do not happen.

**Reply by dudicoco** [reply]
Thanks, you're right after commenting out `eval "$(/opt/homebrew/bin/brew shellenv)"` in my `.zshrc` I get the same error when running nvim normally.
How can I make kitty source my shell rc files? Is there a reason why it doesn't do it when using the `launch` command?

**Comment by kovidgoyal** [comment]
launch runs the program you ask it to run. If you want it to run your
shell ask it to do so and have your shell run nvim in turn.

**Reply by dudicoco** [reply]
I've also tried `zsh -c nvim` with the same result.

**Reply by tg0h** [reply]
looks like your neovim is missing some env variables that your zshrc is setting up...

i had a similar problem with my lsp not working when launching nvim from kitty.

- i added this to my kitty config ... 

```bash
# add node to the PATH so that nvim launched by kitty can find the typescript lang server
env PATH=$PATH:/Users/tim/.local/share/nvm/versions/node/v16.18.0/bin

# i use nvim mason to install my lang servers and they are stored here
env PATH=$PATH:/Users/tim/.local/share/nvim/mason/bin
```

- i also have a lua script that i run from my nvim init lua that sources all the env variables in my .env file

```lua
function M.load_env()
  -- Load environment variables from .env file
  local env_file = os.getenv('XDG_CONFIG_HOME') .. '/secret/env'
  local f = io.open(env_file, 'r')
  if f then
    for line in f:lines() do
     -- if the line looks like export SOME_ENV_VAR=some_value ... then put it in vim.env 
      local _, var, val = line:match('(export )([^=]+)=(.*)')
      if var and val then vim.env[var] = val end
    end
    f:close()
  end
end

return M
```

**Comment by kovidgoyal** [comment]
On Wed, Dec 13, 2023 at 12:35:49PM -0800, dudicoco wrote:
> I've also tried `zsh -c nvim` with the same result.

You need zsh -il -c nvim

Read the zsh ivocation manpage.

**Reply by dudicoco** [reply]
that works, thanks!
